akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -99,7 +99,7 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
       }
 
     def renderContentLength(contentLength: Long) =
-      if (method.isEntityAccepted) r ~~ `Content-Length` ~~ contentLength ~~ CrLf else r
+      if (method.isEntityAccepted && (contentLength > 0 || !method.isSafe)) r ~~ `Content-Length` ~~ contentLength ~~ CrLf else r
```

Comment:
Safe methods usually don't have a body, hence no need for setting `Content-Length: 0`.